### PR TITLE
1237 add send target support to marker authorizations

### PR DESCRIPTION
**Pull request by nullpointer0x00**
## Description

Adds list of allowed addresses to send to with the marker transfer authz. 

NOTE: Ran `make proto-all` which updated a reward module documentation.

closes: #1237

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -37,12 +37,69 @@ Ref: https://keepachangelog.com/en/1.0.0/
...
+
+* Updated Cosmos-SDK to [v0.46.6-pio-1](https://github.com/provenance-io/cosmos-sdk/blob/v0.46.6-pio-1/RELEASE_NOTES.md) (from v0.46.4-pio-1) [PR 1235](https://github.com/provenance-io/provenance/pull/1235).
+* Alias the `config unpack` command to `config update`. It can be used to update config files to include new fields [PR 1233](https://github.com/provenance-io/provenance/pull/1233).
+* When loading the unpacked configs, always load the defaults before reading the files (instead of only loading the defaults if the file doesn't exist) [PR 1233](https://github.com/provenance-io/provenance/pull/1233).
+
+### Full Commit History
+
+* https://github.com/provenance-io/provenance/compare/v1.13.0-rc4...v1.13.0-rc5
+* https://github.com/provenance-io/provenance/compare/v1.12.2...v1.13.0-rc5
+
```
Comment (SpicyLemon) [inline CHANGELOG.md:58]: Bad merge? Lots of changelog additions here that shouldn't be here.

File: x/marker/spec/11_authorization.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Authorization
```
Comment (SpicyLemon) [inline x/marker/spec/11_authorization.md:1]: YAAAAASSSSS!!! (documentation) ❤️‍🔥

**Review by SpicyLemon — COMMENTED** [review]
Just the changelog, the rest looks good!

File: x/marker/client/cli/tx.go [hunk 3]
```diff
@@ -695,10 +706,24 @@ func GetCmdGrantAuthorization() *cobra.Command {
 	}
 	flags.AddTxFlagsToCmd(cmd)
 	cmd.Flags().String(FlagTransferLimit, "", "The total amount an account is allowed to tranfer on granter's behalf")
+	cmd.Flags().StringSlice(FlagAllowList, []string{}, "Allowed addresses grantee is allowed to send restricted coins separated by ,")
 	cmd.Flags().Int64(FlagExpiration, time.Now().AddDate(1, 0, 0).Unix(), "The Unix timestamp. Default is one year.")
 	return cmd
 }
 
+// bech32toAccAddresses returns []AccAddress from a list of Bech32 string addresses.
+func bech32toAccAddresses(accAddrs []string) ([]sdk.AccAddress, error) {
```
Comment (arnabmitra) [inline x/marker/client/cli/tx.go:715]: sucks that the same code exists in authz module, but i guess it's ok, would have been nice if the sdk made it re-usable

File: x/marker/types/errors.go [hunk 4]
```diff
@@ -12,4 +12,5 @@ var (
 	ErrInvalidMarkerStatus     = cerrs.Register(ModuleName, 5, "invalid marker status")
 	ErrAccessTypeNotGranted    = cerrs.Register(ModuleName, 6, "access type not granted")
 	ErrMarkerNotFound          = cerrs.Register(ModuleName, 7, "marker not found")
+	ErrDuplicateEntry          = cerrs.Register(ModuleName, 8, "duplicate entry")
```
Comment (arnabmitra) [inline x/marker/types/errors.go:15]: duplicate entry in allowed list ..maybe .for more precision

File: x/marker/types/authz_test.go [hunk 5]
```diff
@@ -54,3 +54,42 @@ func TestMarkerTransferAuthorization(t *testing.T) {
...
+			NewMarkerTransferAuthorization(sdk.NewCoins(coin500), []sdk.AccAddress{}),
+			"",
+		},
+		{
+			"valid msg without non-empty allow list",
+			NewMarkerTransferAuthorization(sdk.NewCoins(coin500), []sdk.AccAddress{addr1}),
+			"",
+		},
+		{
+			"invalid msg without duplicate allow list",
```
Comment (arnabmitra) [inline x/marker/types/authz_test.go:77]: i think should be `with` not without

File: x/marker/types/authz_test.go [hunk 6]
```diff
@@ -54,3 +54,42 @@ func TestMarkerTransferAuthorization(t *testing.T) {
...
+func TestMarkerTransferAuthorizationValidateBasic(t *testing.T) {
+	addr1, _ := sdk.AccAddressFromBech32("cosmos1sh49f6ze3vn7cdl2amh2gnc70z5mten3y08xck")
+
+	cases := []struct {
+		name     string
+		msg      *MarkerTransferAuthorization
+		errorMsg string
+	}{
+		{
+			"valid msg without empty allow list",
```
Comment (arnabmitra) [inline x/marker/types/authz_test.go:67]: `with` here too ..i think

File: x/marker/types/errors.go (hunk 4 above)
Comment (nullpointer0x00) [reply x/marker/types/errors.go:15]: Was thinking of keeping it more generic for possible usages else where...

File: x/marker/types/authz_test.go (hunk 5 above)
Comment (nullpointer0x00) [reply x/marker/types/authz_test.go]: fixed

File: x/marker/types/authz_test.go (hunk 6 above)
Comment (nullpointer0x00) [reply x/marker/types/authz_test.go]: fixed

**Comment by arnabmitra** [comment]
lgtm overall!..if you fix the change log i think it's good to merge

File: x/marker/types/errors.go (hunk 4 above)
Comment (nullpointer0x00) [reply x/marker/types/errors.go:15]: I ended up using the Wrap on the error with more detail at the point it it hit: `ErrDuplicateEntry.Wrap("all allow list addresses must be unique")`

File: CHANGELOG.md (hunk 1 above)
Comment (nullpointer0x00) [reply CHANGELOG.md]: Yes, bad merge. Thanks!

File: x/marker/spec/11_authorization.md (hunk 2 above)
Comment (nullpointer0x00) [reply x/marker/spec/11_authorization.md:1]: Yes, I have been more strict on missing specs.

**Review by arnabmitra — APPROVED** [review]

**Review by SpicyLemon — APPROVED** [review]

**Review by arnabmitra — APPROVED** [review]

**Pull request merged**